presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -73,6 +73,11 @@ public class OperatorContext
     private final CounterStat outputDataSize = new CounterStat();
     private final CounterStat outputPositions = new CounterStat();
 
+    // Number of NULL elements in hash table
+    private final AtomicLong hashTableNullCount = new AtomicLong();
+    // Number of elements in hash table
+    private final AtomicLong hashTableElementCount = new AtomicLong();
```

Comment:
why are the types different? long here and double above?

Reply:
The counter here are used to record the actual number from query execution. 
All estimated statistics are double.
With HBO, we are storing the actual number as estimates for future use, hence long here and double in estimates.